pkg/values/list.go

```diff
@@ -34,39 +34,46 @@ func (l *List) proto() *pb.Value {
 
 func (l *List) Unwrap() (any, error) {
 	nl := []any{}
-	for _, v := range l.Underlying {
-		uv, err := Unwrap(v)
-		if err != nil {
-			return nil, err
-		}
-
-		nl = append(nl, uv)
-	}
-
-	return nl, nil
+	return nl, l.UnwrapTo(&nl)
 }
 
 func (l *List) UnwrapTo(to any) error {
 	val := reflect.ValueOf(to)
-	if val.Kind() != reflect.Ptr && reflect.Indirect(val).Kind() != reflect.Slice {
-		return fmt.Errorf("cannot unwrap to type %T", to)
+	if val.Kind() != reflect.Pointer {
+		return fmt.Errorf("cannot unwrap to non-pointer type %T", to)
 	}
 
 	if val.IsNil() {
 		return fmt.Errorf("cannot unwrap to nil pointer: %+v", to)
 	}
 
-	newList := reflect.New(reflect.Indirect(val).Type()).Elem()
-	for _, el := range l.Underlying {
-		newEl := reflect.New(reflect.Indirect(val).Type().Elem()).Elem()
-		ptrEl := newEl.Addr().Interface()
-		err := el.UnwrapTo(ptrEl)
+	ptrVal := reflect.Indirect(val)
+	switch ptrVal.Kind() {
+	case reflect.Slice:
+		newList := reflect.New(reflect.Indirect(val).Type()).Elem()
+		for _, el := range l.Underlying {
+			newEl := reflect.New(reflect.Indirect(val).Type().Elem()).Elem()
+			ptrEl := newEl.Addr().Interface()
+			err := el.UnwrapTo(ptrEl)
+			if err != nil {
+				return err
+			}
+			newList = reflect.Append(newList, reflect.Indirect(reflect.ValueOf(ptrEl)))
+		}
+		reflect.Indirect(val).Set(newList)
+		return nil
+	default:
+		dl := []any{}
+		err := l.UnwrapTo(&dl)
```

Comment:
I don't understand this - if "to" is not a slice, why are we still trying to unwrap rather than fail?

Reply:
This handles the case where you pass in an uninstantiated `any` pointer.

Without that this test case fails: https://github.com/smartcontractkit/chainlink-common/pull/397/files#diff-4c5e50901d7b6e0f0668549fa0a61acb044d57a42a82e2020fbaf601934afbecR67